litellm/llms/notdiamond.py

```diff
@@ -0,0 +1,262 @@
+        )
+    headers = {
+        "Authorization": "Bearer " + api_key,
+        "accept": "application/json",
+        "content-type": "application/json",
+        "User-Agent": f"litellm/{version}",
+    }
+    return headers
+
+
+def get_litellm_model(response: dict) -> str:
+    nd_provider = response["providers"][0]["provider"]
+    nd_model = response["providers"][0]["model"]
+    nd_provider_model = f"{nd_provider}/{nd_model}"
+    litellm_model = ND2LITELLM[nd_provider_model]
+    return litellm_model
+
+
+def update_litellm_params(litellm_params: dict):
+    """
+    Create a new litellm_params dict with non-default litellm_params from the original call, custom_llm_provider and api_base
+    """
+    new_litellm_params = dict()
+    for k, v in litellm_params.items():
+        # all litellm_params have defaults of None or False, except force_timeout
+        if (k == "force_timeout" and v != 600) or v:
+            new_litellm_params[k] = v
+    if "custom_llm_provider" in new_litellm_params:
+        del new_litellm_params["custom_llm_provider"]
+    if "api_base" in new_litellm_params:
+        del new_litellm_params["api_base"]
+    if "api_key" in new_litellm_params:
+        del new_litellm_params["api_key"]
+    return new_litellm_params
+
+
+def completion(
+    model: str,
+    messages: list,
+    api_base: str,
+    model_response: ModelResponse,
+    print_verbose: Callable,
+    encoding,
+    api_key,
+    logging_obj,
+    optional_params=None,
+    litellm_params=None,
+    logger_fn=None,
+):
+    headers = validate_environment(api_key)
+    completion_url = api_base
+
+    ## Load Config
+    config = litellm.NotDiamondConfig.get_config()
+    for k, v in config.items():
+        if k not in optional_params:
+            optional_params[k] = v
+
+    # separate ND optional params from litellm optional params
+    nd_params = [
+        "llm_providers",
+        "tools",
+        "max_model_depth",
+        "tradeoff",
+        "preference_id",
+        "hash_content",
+    ]
+    selected_model_params = dict()
+    for k, v in optional_params.items():
+        if k not in nd_params:
+            selected_model_params[k] = v
+    if "tools" in optional_params:
+        selected_model_params["tools"] = optional_params["tools"]
+    # remove any optional params that are not in the ND params
+    optional_params = {k: v for k, v in optional_params.items() if k in nd_params}
+
+    data = {
+        "messages": messages,
+        **optional_params,
+    }
+
+    ## LOGGING
+    logging_obj.pre_call(
+        input=messages,
+        api_key=api_key,
+        additional_args={
+            "complete_input_dict": data,
+            "headers": headers,
+            "api_base": completion_url,
+        },
+    )
+
+    ## MODEL SELECTION CALL
+    nd_response = requests.post(
```

Comment:
in fact it might be better to have it above loadbalancing - as a pre-call hook, so that if your api selects 'gpt-3.5-turbo', the call is made across the loadbalanced group - https://docs.litellm.ai/docs/proxy/call_hooks


you can:
- check if the incoming request is of call_type == "acompletion"
- then, send the data to your api endpoint (data will be the request body for a `/chat/completion` call) 
- then, replace the data["model"] with your selected model name